app/controllers/posts_controller.rb

```diff
@@ -1,12 +1,39 @@
 class PostsController < ApplicationController
   def index
-    @user = User.find(params[:user_id])
-    @posts = @user.posts
+    @user = User.find_by(id: params[:user_id])
+    @posts = current_user.posts.includes(:comments)
   end
 
   def show
-    @post = Post.find(params[:id])
+    @user = User.find_by(id: params[:user_id])
+    @post = Post.find_by(id: params[:id])
     @comments = @post.comments
     @likes = @post.likes
+    @new_comment = @post.comments.new
+  end
+
+  def new
+    @user = current_user
```

Comment:
[OPTIONAL] In the `new` method, there is no need for the instant variable `@user`.